src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/LightBoxMercuryComponentObject.java

```diff
@@ -73,5 +94,172 @@ public void verifyImageWasChanged(String imageOnePath, String imageTwoPath) {
+			touchAction.swipeFromPointToPoint(startX, 50, endX, 50, duration, waitAfter);
+			nextImageSrc = getCurrentImagePath();
+			if (!nextImageSrc.contains(currentImageSrc)) {
+				imageChanged = true;
+				break;
+			}
+		}
+		if (imageChanged) {
+			PageObjectLogging.log("verifySwiping", "Swiping to " + direction + " works", true);
+		} else {
+			PageObjectLogging.log("verifySwiping", "Swiping to " + direction + " doesn't work", false);
+		}
+	}
+	
+	private boolean verifyChangesInUI (boolean lastDisplayVisibility) {
+		String lightboxHeaderDisplayValue = lightboxHeader.getCssValue("display");
+		String lightboxFooterDisplayValue = lightboxFooter.getCssValue("display");
+		boolean lightboxHeaderDisplay;
+		boolean lightboxFooterDisplay;
+		if (lightboxHeaderDisplayValue.contains("none")) {
+			lightboxHeaderDisplay = false;
+		} else {
+			lightboxHeaderDisplay = true;
+		}
+		if (lightboxFooterDisplayValue.contains("none")) {
+			lightboxFooterDisplay = false;
+		} else {
+			lightboxFooterDisplay = true;
+		}
+		if (lightboxHeaderDisplay == lastDisplayVisibility) {
+			PageObjectLogging.log("verifyChangesInUI", "Lightbox header visibility didn't changed", false);
+		} else {
+			PageObjectLogging.log("verifyChangesInUI", "Lightbox header visibility changed", true);
+		}
+		if (lightboxFooterDisplay == lastDisplayVisibility) {
+			PageObjectLogging.log("verifyChangesInUI", "Lightbox footer visibility didn't changed", false);
+		} else {
+			PageObjectLogging.log("verifyChangesInUI", "Lightbox footer visibility changed", true);
+		}
+		return !lastDisplayVisibility;
+	}
+	
+	public void verifyVisibilityUI (PerformTouchAction touchAction) {
+		String LightboxHeaderDisplayValue = lightboxHeader.getCssValue("display");
+		String LightboxFooterDisplayValue = lightboxFooter.getCssValue("display");
+		int duration = 500;
+		int waitAfter = 5000;
+		boolean lastDisplayVisibility = true;
+		if (LightboxHeaderDisplayValue.contains(LightboxFooterDisplayValue)) {
+			if (LightboxHeaderDisplayValue == "none") {
+				PageObjectLogging.log("verifyVisibilityUI", "Lightbox header and footer are not visible", true);
+				lastDisplayVisibility = false;
+			} else {
+				PageObjectLogging.log("verifyVisibilityUI", "Lightbox header and footer are visible", true);
+				lastDisplayVisibility = true;
+			}
+		} else {
+			PageObjectLogging.log("verifyVisibilityUI", "Visibility of lightbox header is different than lightbox footer", false);
+		}
+		touchAction.tapOnPointXY(50, 50, duration, waitAfter);
+		lastDisplayVisibility = verifyChangesInUI(lastDisplayVisibility);
+		touchAction.tapOnPointXY(50, 50, duration, waitAfter);
+		lastDisplayVisibility = verifyChangesInUI(lastDisplayVisibility);
+	}
+	
+	public void verifyTappingOnImageEdge (PerformTouchAction touchAction, String edge) {
+		String currentImageSrc = getCurrentImagePath();
+		String nextImageSrc;
+		int pointX = 25;
+		int duration = 500;
+		int waitAfter = 5000;
+		if (edge == "right") {
```

Comment:
if doing a String comparison, it needs to be used with .equals()
I would suggest putting the String on the left side to avoid NullPointerException